olp-cpp-sdk-core/src/cache/ProtectedData.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "ProtectedData.h"
+
+#include <algorithm>
+#include <string>
+#include <utility>
+
+#include "olp/core/logging/Log.h"
+
+namespace {
+constexpr auto kLogTag = "ProtectedData";
+}  // namespace
+
+namespace olp {
+namespace cache {
+
+ProtectedData::ProtectedData() {}
+
+ProtectedData::~ProtectedData() {}
+
+bool ProtectedData::ReadListOfProtectedKeys(KeyValueCache::ValueTypePtr value) {
+  auto result = true;
+  std::string str;
+  for (const auto& el : *value) {
+    if (el == '\0' && !str.empty()) {
+      protected_data_.insert(str);
+      str.clear();
+    } else {
+      str += el;
+    }
+  }
+  return result;
+}
+
+KeyValueCache::ValueTypePtr ProtectedData::WriteListOfProtectedKeys() {
+  auto value = std::make_shared<std::vector<unsigned char>>();
+  auto size = 0;
+  for (const auto& key : protected_data_) {
+    size += key.length() + 1;
+  }
+  if (size > 0) {
+    value->reserve(size);
+
+    for (const auto& key : protected_data_) {
+      value->insert(value->end(), key.begin(), key.end());
+      if (key.back() != '\0') {
+        value->emplace_back('\0');
+      }
+    }
+  }
+  return value;
+}
+
+bool ProtectedData::Protect(
+    const DefaultCache::KeyListType& keys,
+    const ProtectedKeyChanged& change_key_to_protected) {
+  for (const auto& key : keys) {
+    // find key or prefix
+    auto hint = protected_data_.lower_bound(key);
+    if (hint != protected_data_.end()) {
+      // if hint is prefix for this key, ignore this key, it is allready
+      // protected
+      if (IsPrefix(*hint, key)) {
+        continue;
+      }
+      // if key is prefix for hint key, remove hint and add prefix
+      while (hint != protected_data_.end() && IsPrefix(key, *hint)) {
+        hint = protected_data_.erase(hint);
+      }
+    }
+
+    // remove keys from lru and memory cache
```

Comment:
This is irrelevant here and should not be covered by this class, means remove comment or change to something more generic like: "Notify listener that keys have changed".
Also, why do you need to remove something from the memory cache?